app/controllers/users.py

```diff
@@ -1239,3 +1281,88 @@ def post(self, user_id):
                 status='fail',
                 message=str(err)
             ), 500
+
+
+class UserFollowView(Resource):
+    @ jwt_required
+    def post(self, user_id):
+        current_user_id = get_jwt_identity()
+        current_user = User.get_by_id(current_user_id)
+        user = User.get_by_id(user_id)
+
+        if not user:
+            return dict(status='fail', message=f'User with id {user_id} not found'), 404
+
+        if user == current_user:
+            return dict(status='fail', message='You cannot follow yourself'), 400
+
+        if user in current_user.followed:
+            return dict(status='fail', message=f'You are already following user with id {user_id}'), 409
+
+        current_user.followed.append(user)
+        saved_user = current_user.save()
+
+        if not saved_user:
+            return dict(status='fail', message='Internal Server Error'), 500
+
+        return dict(
+            status='success',
+            message=f'You are now following user with id {user_id}'
+        ), 201
+
+    @ jwt_required
+    def get(self, user_id):
+        user = User.get_by_id(user_id)
+        user_schema = UserSchema(many=True)
+
+        followed = user.followed
+        users_data, errors = user_schema.dumps(followed)
+
+        if errors:
+            return dict(status='fail', message=errors), 400
+
+        return dict(
+            status='success',
+            data=dict(following=json.loads(users_data))
+        ), 200
+
+    @ jwt_required
+    def delete(self, user_id):
+        current_user_id = get_jwt_identity()
+        current_user = User.get_by_id(current_user_id)
+        user = User.get_by_id(user_id)
+
+        if not user:
+            return dict(status='fail', message=f'User with id {user_id} not found'), 404
+
+        if user not in current_user.followed:
+            return dict(status='fail', message=f'You are not following user with id {user_id}'), 409
+
+        current_user.followed.remove(user)
+        saved_user = current_user.save()
+
+        if not saved_user:
+            return dict(status='fail', message='Internal Server Error'), 500
+
+        return dict(
+            status='success',
+            message=f'You have unfollowed user with id {user_id}'
+        ), 200
+
+
+class UserFollowersView(Resource):
```

Comment:
Legit work @Mubangizi , Just wondering if we can have a delete route here such that a user is able to delete a follower incase of inconvieniences . Something like this :

```suggestion
class UserFollowersView(Resource):
@ jwt_required
    def delete(self, user_id):
        current_user_id = get_jwt_identity()
        current_user = User.get_by_id(current_user_id)
        user = User.get_by_id(user_id)

        if not user:
            return dict(status='fail', message=f'User with id {user_id} not found'), 404

        if user not in current_user.followers:
            return dict(status='fail', message=f'You are not followed by user with id {user_id}'), 409

        current_user.followers.remove(user)
        saved_user = current_user.save()
        if not saved_user:
            return dict(status='fail', message='Internal Server Error'), 500

        return dict(
            status='success',
            message=f'You have deleted user {user_id} from your followers'
        ), 200
```

Reply:
I don't think You are supposed to delete a follower. Maybe you can block them but not force them to unfollow you